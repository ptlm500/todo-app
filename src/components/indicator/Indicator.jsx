import React from 'react';
import PropTypes from 'prop-types';
import styled, { css, keyframes } from 'styled-components';
import { theme } from '../../styles';

export const circlePulse = () => keyframes`
0% {
  opacity: 1;
}
50% {
  opacity: 0.5;
}
100%{
  opacity: 1;
}
`;

const StyledIndicator = styled.svg`
  display: inline-block;
  margin: auto;
  width: 12px;
  height: 12px;
`;

const StyledInnerCircle = styled.circle`
  animation: ${props => props.disabled || !props.pulsing ? 'none' :
    css`${circlePulse()} infinite 4s ease-in-out;`};
  fill: ${props => props.disabled ? theme.disabled : props.fill};
  width: 100;
  height: 100;
  transition: ${theme.shortTransition};
`;


export default function Indicator(props) {
  return (
    <StyledIndicator
      viewBox="0 0 100 100"
    >
      <StyledInnerCircle
        cx="50"
        cy="50"
        r="50"
        {...props}
      />
    </StyledIndicator>
  );
}

Indicator.propTypes = {
  disabled: PropTypes.bool,
  fill: PropTypes.string.isRequired,
  pulsing: PropTypes.bool
};
