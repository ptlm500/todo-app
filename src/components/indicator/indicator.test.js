import React from 'react';
import renderer from 'react-test-renderer';
import Indicator, { circlePulse } from './Indicator';
import 'jest-styled-components';

describe('Indicator', () => {
  it('renders an indicator', () => {
    const tree = renderer.create(<Indicator fill={'#fff'} />);
    expect(tree).toMatchSnapshot();
  });

  it('renders a disabled indicator', () => {
    const tree = renderer.create(<Indicator fill={'#fff'} disabled />);
    expect(tree).toMatchSnapshot();
  });
});

describe('pulsing Indicator', () => {
  it('renders a pulsing indicator', () => {
    const tree = renderer.create(<Indicator pulsing fill={'#fff'} />);
    expect(tree).toMatchSnapshot();
  });

  it('renders a disabled pulsing indicator', () => {
    const tree = renderer.create(<Indicator pulsing fill={'#fff'} disabled />);
    expect(tree).toMatchSnapshot();
  });
});

describe('circlePulse', () => {
  it('matches the snapshot', () => {
    expect(circlePulse()).toMatchSnapshot();
  });
});
