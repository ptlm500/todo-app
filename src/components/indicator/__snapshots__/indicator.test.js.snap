// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Indicator renders a disabled indicator 1`] = `
.c0 {
  display: inline-block;
  width: 12px;
  height: 12px;
}

.c1 {
  -webkit-animation: none;
  animation: none;
  fill: #BDBDBD;
  width: 100;
  height: 100;
  -webkit-transition: 0.1s;
  transition: 0.1s;
}

<svg
  className="c0"
  viewBox="0 0 100 100"
>
  <circle
    className="c1"
    cx="50"
    cy="50"
    disabled={true}
    fill="#fff"
    r="50"
  />
</svg>
`;

exports[`Indicator renders an indicator 1`] = `
.c0 {
  display: inline-block;
  width: 12px;
  height: 12px;
}

.c1 {
  -webkit-animation: none;
  animation: none;
  fill: #fff;
  width: 100;
  height: 100;
  -webkit-transition: 0.1s;
  transition: 0.1s;
}

<svg
  className="c0"
  viewBox="0 0 100 100"
>
  <circle
    className="c1"
    cx="50"
    cy="50"
    fill="#fff"
    r="50"
  />
</svg>
`;

exports[`circlePulse matches the snapshot 1`] = `
Keyframes {
  "id": "sc-keyframes-ehXoqr",
  "inject": [Function],
  "name": "ehXoqr",
  "stringifyArgs": Array [
    "
0% {
  opacity: 1;
}
50% {
  opacity: 0.5;
}
100%{
  opacity: 1;
}
",
    "ehXoqr",
    "@keyframes",
  ],
  "toString": [Function],
}
`;

exports[`pulsing Indicator renders a disabled pulsing indicator 1`] = `
.c0 {
  display: inline-block;
  width: 12px;
  height: 12px;
}

.c1 {
  -webkit-animation: none;
  animation: none;
  fill: #BDBDBD;
  width: 100;
  height: 100;
  -webkit-transition: 0.1s;
  transition: 0.1s;
}

<svg
  className="c0"
  viewBox="0 0 100 100"
>
  <circle
    className="c1"
    cx="50"
    cy="50"
    disabled={true}
    fill="#fff"
    r="50"
  />
</svg>
`;

exports[`pulsing Indicator renders a pulsing indicator 1`] = `
.c0 {
  display: inline-block;
  width: 12px;
  height: 12px;
}

.c1 {
  -webkit-animation: ehXoqr infinite 4s ease-in-out;
  animation: ehXoqr infinite 4s ease-in-out;
  fill: #fff;
  width: 100;
  height: 100;
  -webkit-transition: 0.1s;
  transition: 0.1s;
}

<svg
  className="c0"
  viewBox="0 0 100 100"
>
  <circle
    className="c1"
    cx="50"
    cy="50"
    fill="#fff"
    r="50"
  />
</svg>
`;
